### ISSUE TITLE
Is there a way to get the number of pages without loading the entire pdf document?

### ISSUE THREAD

**Issue by swifttarrow** [wojtekmaj/react-pdf#350]
**Before you start - checklist**

 - [X] I have read documentation in README
 - [X] I have checked [sample](https://github.com/wojtekmaj/react-pdf/tree/master/sample) and [test](https://github.com/wojtekmaj/react-pdf/tree/master/test) suites to see real life basic implementation
 - [X] I have checked if this question is not already asked

**What are you trying to achieve? Please describe.**
For v1 of our pdf viewer (built on top of your library), we did not build any virtualization mechanism. As a result, the browser gets unresponsive as the number of pages increases. We want to show an error message if the pdf exceeds a certain number of pages. Ideally, we'd like to it before loading the entire pdf document; is this possible?

**Describe solutions you've tried**
Read through the README and saw that the file prop mentions a PDFDataRangeTransport param. Coupled with the FAQ from PDF.js, it sounds like the number of pages might be stored somewhere at the end of the file's bytestream (?) 

Thanks in advance!

**Comment by wojtekmaj** [comment]
TL;DR I don't know and I certainly did not implement something like this using React-PDF

React-PDF uses Mozilla's PDF.js to load the document. Loading the document is what gives us info on the document, including the number of pages, which we later pass to `onLoadSuccess` callback.

I suggest following #94 for solutions on rendering large PDFs.
